### PR TITLE
[Apache Tomcat] Add integration package with memory data stream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (Memory metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] Verify if field mapping is correct in the data stream template.
- [x] Verification of data in visualisation after enabling TSDB flag in kibana
- [x] Verification of the count of documents (before & after TSDB enablement) in Discover Interface

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/apache_tomcat) directory.
- Run the following command to run tests. elastic-package test

## Screenshots
apache_tomcat-memory-dashboard

## Meta issue
- #5200

## Related PRs
- #5424
- #5926
- #5883
- #5983
- #6003
- #6573
- #6596
- #6609

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Review by milan-elastic — APPROVED** [review]

File: packages/apache_tomcat/data_stream/memory/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        default:
+          - forwarded
+          - apache_tomcat-memory
+      - name: processors
+        type: yaml
+        title: Processors
+        multi: false
+        required: false
+        show_user: false
+        description: Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
```
Comment (ishleenk17) [inline packages/apache_tomcat/data_stream/memory/manifest.yml:30]: Please link to the Elastic Agent Processors link https://www.elastic.co/guide/en/fleet/current/elastic-agent-processor-configuration.html

File: packages/apache_tomcat/data_stream/memory/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            - name: collection
+              type: group
+              fields:
+                - name: count
+                  type: long
+                  description: The cumulative number of invoked garbage collections since the start of the server.
+                  metric_type: gauge
+                - name: time.ms
+                  type: long
+                  description: The amount of time (in milliseconds) that garbage collection took during the collection interval.
```
[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/apache_tomcat/data_stream/memory/fields/fields.yml]: Updated, thanks!

File: packages/apache_tomcat/data_stream/memory/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                  metric_type: gauge
+                  unit: ms
+            - name: valid
+              type: long
+              description: Valid Garbage collection.
```
[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/apache_tomcat/data_stream/memory/fields/fields.yml]: Updated, thanks!

File: packages/apache_tomcat/data_stream/memory/manifest.yml (hunk 1 above)
Comment (harnish-elastic) [reply packages/apache_tomcat/data_stream/memory/manifest.yml]: Updated, thanks!

**Comment by ishleenk17** [comment]
Looks good!
I would wait for Agi/Richa's feedback for tsdb perspective. @agithomas @ritalwar .
@SubhrataK : Any feedback for dashboards ?

File: packages/apache_tomcat/data_stream/memory/fields/ecs.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+- external: ecs
+  name: agent.id
+  dimension: true
+- external: ecs
+  name: cloud.instance.id
+  dimension: true
+- external: ecs
+  name: cloud.project.id
```
Comment (agithomas) [inline packages/apache_tomcat/data_stream/memory/fields/ecs.yml:8]: Please refer [here](https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359) for the latest common dimension fields. Please update the mappings for all datastreams.

Comment (harnish-elastic) [reply packages/apache_tomcat/data_stream/memory/fields/ecs.yml]: Updated, thanks!

File: packages/apache_tomcat/data_stream/memory/fields/fields.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        - name: gc
+          type: group
+          fields:
+            - name: collection
+              type: group
+              fields:
+                - name: count
+                  type: long
+                  description: The cumulative number of invoked garbage collections since the start of the server.
+                  metric_type: gauge
```
[the remark this answers is not in the document]
Comment (harnish-elastic) [reply packages/apache_tomcat/data_stream/memory/fields/fields.yml]: Updated, thanks!

**Review by agithomas — APPROVED** [review]
LGTM considering only TSDB.

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Comment by harnish-elastic** [comment]
As discussed with @SubhrataK regarding the memory dashboard, re-arrange the order of visualizations, updated metrics chart visualizations and updated color for Garbage collection over time chart for better readability.
memory-dashboard

**Review by SubhrataK — APPROVED** [review]
Looks good.

**Pull request merged**